frontend/src/components/common/TabNavigation.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import Tab, { SELECTED, UNSELECTED } from "./Tab";
+
+class TabNavigation extends Component {
+  static propTypes = {
+    tabSpecs: PropTypes.array.isRequired,
+    currentTab: PropTypes.number.isRequired
+  };
+
+  state = {
+    tabSpecs: this.props.tabSpecs,
+    currentTab: this.props.currentTab,
+    currentBody: this.props.tabSpecs[this.props.currentTab].body
+  };
+
+  selectTab(id) {
+    let tempSpecs = this.state.tabSpecs;
+    tempSpecs[this.state.currentTab].selected = UNSELECTED;
```

Comment:
Instead of constants defined in another file, I think you should just use booleans directly everywhere. It makes perfect sense for `selected = true;` and to me feels harder to read to have `selected = UNSELECTED;`. If there were more than two options, or the options were not just true/false, a constant would make sense.

Reply:
completed